src/internal/plugins/helpers/runtime-content-wrapper.tsx

```diff
@@ -2,18 +2,27 @@
 // SPDX-License-Identifier: Apache-2.0
 import React, { useEffect, useRef } from 'react';
 
+type VisibilityCallback = (isVisible: boolean) => void;
+
 interface RuntimeContentWrapperProps {
-  mountContent: (container: HTMLElement) => void;
+  mountContent: (container: HTMLElement, visibilityCallback?: (callback: VisibilityCallback) => () => void) => void;
   unmountContent: (container: HTMLElement) => void;
+  registerVisibilityCallback?: (callback: VisibilityCallback) => () => void;
 }
 
-export function RuntimeContentWrapper({ mountContent, unmountContent }: RuntimeContentWrapperProps) {
+export function RuntimeContentWrapper({
+  mountContent,
+  unmountContent,
+  registerVisibilityCallback,
+}: RuntimeContentWrapperProps) {
   const ref = useRef<HTMLDivElement>(null);
 
   useEffect(() => {
     const container = ref.current!;
-    mountContent(container);
-    return () => unmountContent(container);
+    mountContent(container, registerVisibilityCallback);
+    return () => {
+      unmountContent(container);
+    };
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
```

Comment:
I think, this can be simplified

```js
function RuntimeContentWrapper({mountContent, unmountContent, isVisible}) {
   const visibilityChangeCallback = useRef();

  useEffect(() => {
    const container = ref.current!;
    mountContent(container, {
      onVisibilityChange: cb => visibilityChangeCallback.current = cb
    });
    return () => unmountContent(container);
  }, []);

   useEffect(() => {
      visibilityChangeCallback.current?.(isVisible)
   }, [isVisible]);
   
   ... other code
}
```

In this setup, you do not need `VisibilityStateManager` because React will do all state management for you